### PR TITLE
feature: reorganize and split tests

### PR DESCRIPTION
**Pull request by h4sh3d**
Split tests into more files, each test start with `{file}_` e.g. `swap_` in `swap.rs` or `monero_` in `monero.rs`. Test are run the same way as before `cargo tests {monero|bitcoin|swap|...} ...`.

List of tests category/files:
- swap
- monero
- bitcoin
- grpc
- cli

`utils` module should still receive more code from root test files and should later be improved to help stabilizing the tests.

#### This PR aim to not alter tests behavior, just to reorganize so we can work on test improvments independently without conflict later

### CODE REVIEW

**Review by TheCharlatan — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -53,9 +53,25 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        test: [bitcoin, monero, swap]
```
Comment (TheCharlatan) [inline .github/workflows/test.yml:56]: The timeout-minutes should be significantly reduced again. I think we can go back to the old 45 min limit.

Comment (h4sh3d) [reply .github/workflows/test.yml:56]: Done 9ecacfbae4a48c5dd20c544274f9b7849c117eab

File: tests/utils/mod.rs [hunk 2]
```diff
@@ -1,55 +1,14 @@
...
-    let node = NodeConfig {
-        host: "localhost".to_string(),
-        port: 123,
-        protocol: None,
-    };
-    let display = node.to_string();
-    assert_eq!(display, "localhost:123");
+pub fn setup_logging() {
+    // !!! Configure RUST_LOG in CI to change this value !!!
+    let env = env_logger::Env::new().default_filter_or("farcaster_node=info");
```
[the remark this answers is not in the document]
Comment (h4sh3d) [reply tests/utils/mod.rs]: Done 9ecacfbae4a48c5dd20c544274f9b7849c117eab

**Comment by h4sh3d** [comment]
I want to fix and upgrade grpc in another PR. Looks like binding on ipv6 might be the problem, but we currently cannot change through config. If ok we can merge it as is

**Comment by TheCharlatan** [comment]
I added a commit with a TODO to disable grpc tests for now.

**Review by TheCharlatan — APPROVED** [review]

**Pull request merged**